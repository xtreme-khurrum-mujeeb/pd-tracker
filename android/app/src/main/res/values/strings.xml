<resources>
    <string name="app_name">pd_tracker</string>
</resources>
