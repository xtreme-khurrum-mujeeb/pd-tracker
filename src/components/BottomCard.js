import React, { Component } from 'react';
import { View, StyleSheet, Image } from 'react-native';

const BottomCard = (props) => {
  return (
      <View style={styles.containerStyle}>

        <View style={styles.boxStyle}>
          {props.children}
        </View>
      </View>
    );
  };

const styles = StyleSheet.create({
  containerStyle: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'flex-end',
    },
  boxStyle: {
    backgroundColor: 'rgba(3,100,87,0.60)',
    width: '100%',
    height: '50%'
  }
});

export default BottomCard;
