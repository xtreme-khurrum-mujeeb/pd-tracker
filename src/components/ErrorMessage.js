import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

const ErrorMessage = ({ message }) => {
  return (
    <View style={styles.errorContainer}>
      <Text style={styles.errorStyle}>
        {message}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  errorContainer: {
    backgroundColor: '#FFAC28',
    flex: 0.20,
    justifyContent: 'center',
    alignItems: 'center'
  },
  errorStyle: {
    color: 'white',
    fontFamily: 'Avenir Next',
    fontSize: 20
  }
});

export default ErrorMessage;
