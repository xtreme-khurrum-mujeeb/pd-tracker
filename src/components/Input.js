import React from 'react';
import { TextInput, View, Text } from 'react-native';

const Input = ({ label, value, onChangeText, placeholder, secureTextEntry }) => {
  const { inputStyle, labelStyle, containerStyle } = styles;

  return (
    <View style={containerStyle}>
      <Text style={labelStyle}>{label}</Text>
      <TextInput
        secureTextEntry={secureTextEntry}
        placeholder={placeholder}
        autoCorrect={false}
        style={inputStyle}
        value={value}
        onChangeText={onChangeText}
        placeholderTextColor="white"
        underlineColorAndroid='transparent'
      />
    </View>
  );
};

const styles = {
  inputStyle: {
    height: 40,
    borderBottomColor: 'white',
    borderBottomWidth: 1,
    color: 'white',
    fontFamily: 'Avenir Next',
    marginTop: '10%'
  },
  labelStyle: {
    fontSize: 18,
    flex: 1
  },
  containerStyle: {
    flex: 1,
    flexDirection: 'column',

  }
};

export default Input;
