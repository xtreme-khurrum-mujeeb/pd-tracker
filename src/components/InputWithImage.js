import React from 'react';
import {
  TextInput,
  View,
  Image,
  StyleSheet
 } from 'react-native';

const InputWithImage = ({ value, onChangeText, placeholder, secureTextEntry, imageSource }) => {
const {
    input,
    inputWrap,
    iconWrap,
    iconStyle
   } = styles;

  return (
    <View style={inputWrap}>
      <View style={iconWrap}>
        <Image
          source={imageSource}
          style={iconStyle}
          resizeMode="contain"
        />
      </View>
        <TextInput
          secureTextEntry={secureTextEntry}
          placeholder={placeholder}
          autoCorrect={false}
          style={input}
          value={value}
          onChangeText={onChangeText}
          placeholderTextColor="#F3EFEF"
          underlineColorAndroid='transparent'
          returnKeyType='next'
        />

    </View>
  );
};

const styles = StyleSheet.create({
  inputWrap: {
    flexDirection: 'row',
    marginVertical: 10,
    height: 40,
    backgroundColor: 'transparent'
  },
  iconWrap: {
    paddingHorizontal: 7,
    alignItems: 'center',
    justifyContent: 'center'
 },
 iconStyle: {
     height: 21,
     width: 21
 },
 input: {
   flex: 1,
   paddingHorizontal: 10,
   borderBottomWidth: 1,
   borderBottomColor: 'white',
   color: 'white',
   fontFamily: 'Avenir Next',
   fontSize: 20,
 },
});

export default InputWithImage;
