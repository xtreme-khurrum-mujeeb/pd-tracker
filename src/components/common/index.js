export * from './Background';
export * from './BottomCard';
export * from './ErrorMessage';
export * from './InputWithImage';
export * from './LargeButton';
export * from './Spinner';
export * from './TopCard';
export * from './ProfileImage';
export * from './IconButton';
export * from './Title';
export * from './Pill';
