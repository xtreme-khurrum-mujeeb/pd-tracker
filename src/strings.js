const strings = {
  invalidCredentials: 'Invalid username or password'
};

export default strings;
