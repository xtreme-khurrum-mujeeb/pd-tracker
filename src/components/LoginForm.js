import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
   View,
   StyleSheet,
   Image,
   Keyboard
 } from 'react-native';
import {
  emailChanged,
  passwordChanged,
  loginUser,
  registerUser
 } from '../actions';

import TopCard from './TopCard';
import strings from '../strings';
import BottomCard from './BottomCard';
import Background from './Background';
import InputWithImage from './InputWithImage';
import LargeButton from './LargeButton';
import ErrorMessage from './ErrorMessage';
import Spinner from './Spinner';

const userIcon = require('../assets/images/user.png');
const passwordIcon = require('../assets/images/padlock.png');

export class LoginForm extends Component {

  static propTypes = {
    emailChanged: PropTypes.func,
    passwordChanged: PropTypes.func,
    loginUser: PropTypes.func,
    registerUser: PropTypes.func,
    email: PropTypes.string,
    password: PropTypes.string,
    error: PropTypes.bool,
    loading: PropTypes.bool
  };

  onEmailChange(text) {
    this.props.emailChanged(text);
  }

  onPasswordChange(text) {
    this.props.passwordChanged(text);
  }

  onLoginPress() {
    const { email, password } = this.props;

    Keyboard.dismiss();

    this.props.loginUser({ email, password });
  }

  onRegisterPress() {
    const { email, password } = this.props;

    this.props.registerUser({ email, password });
  }

  displayErrorMessage() {
    if (this.props.error) {
      return (
        <ErrorMessage message={strings.invalidCredentials} />
      );
    }
  }

  displaySpinner() {
    if (this.props.loading) {
      return (
        <Spinner />
      );
    }
  }

  renderLoginView() {
    return (
      <View style={styles.container}>
        {this.displayErrorMessage()}
        {this.displaySpinner()}
        <View style={styles.wrapper}>
             <InputWithImage
               onChangeText={this.onEmailChange.bind(this)}
               value={this.props.email}
               placeholder="Username"
               imageSource={userIcon}
             />
             <InputWithImage
               secureTextEntry
               onChangeText={this.onPasswordChange.bind(this)}
               value={this.props.password}
               placeholder="Password"
               imageSource={passwordIcon}
               defaultValue="test123"
             />
           <LargeButton
             name='Sign In'
             onPress={this.onLoginPress.bind(this)}
           />
           <LargeButton
             name='Register'
             onPress={this.onRegisterPress.bind(this)}
           />
       </View>
     </View>
  );
}

  render() {
    return (
       <Background>
         <TopCard>
           <View style={styles.topImageContainer}>
             <Image
             source={require('../assets/images/Oval.png')}
             style={styles.topImageStyle}
             />
          </View>
        </TopCard>
         <BottomCard>
           {this.renderLoginView()}
        </BottomCard>
      </Background>
   );
 }
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: 'column',
      justifyContent: 'center'
    },
    wrapper: {
      flex: 1,
      paddingHorizontal: 50,
    },
    topImageContainer: {
      flex: 1,
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center'
    },
    topImageStyle: {
      width: 165,
      height: 165,
      resizeMode: 'contain',
  }

});

const mapStateToProps = state => {
  return {
    email: state.auth.email,
    password: state.auth.password,
    error: state.auth.error,
    loading: state.auth.loading
  };
};

export default connect(mapStateToProps, {
  emailChanged,
  passwordChanged,
  loginUser,
  registerUser
})(LoginForm);
