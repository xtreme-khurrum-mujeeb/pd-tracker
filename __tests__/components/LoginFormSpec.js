import 'react-native';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { LoginForm } from '../../src/components/LoginForm';
import strings from '../../src/strings';

const errorMessage = strings.invalidCredentials;
let initialState;
let subject;
let emailChanged;
let passwordChanged;
let loginUser;
let registerUser;

beforeEach(() => {
  initialState = {
      email: 'test@email.ca',
      password: 'password123',
      error: false,
      loading: false,
  };
});

it('should render the LoginForm', () => {
  const tree = renderer.create(<LoginForm {...initialState} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should pass text in the email field to InputWithImage', () => {
  subject = shallow(<LoginForm {...initialState} />);
  expect(subject.find('InputWithImage').at(0).prop('value')).toContain('test@email.ca');
});

it('should pass in the password field to InputWithImage', () => {
  subject = shallow(<LoginForm {...initialState} />);
  expect(subject.find('InputWithImage').at(1).prop('value')).toContain('password123');
});

it('should be a secureTextEntry on the password field', () => {
  subject = shallow(<LoginForm {...initialState} />);
  expect(subject.find('InputWithImage').at(1).prop('secureTextEntry')).toBeTruthy();
});

it('should not display an error message on intial launch', () => {
  initialState.error = false;
  subject = shallow(<LoginForm {...initialState} />);
  expect(subject.find('ErrorMessage').exists()).toBeFalsy();
});

it('should display an error message with invalid credentials', () => {
  initialState.error = true;
  subject = shallow(<LoginForm {...initialState} />);
  expect(subject.find('ErrorMessage').at(0).prop('message')).toContain(errorMessage);
});

it('should not display a spinner when not in the loading state', () => {
  initialState.loading = false;
  subject = shallow(<LoginForm {...initialState} />);
  expect(subject.find('Spinner').exists()).toBeFalsy();
});

it('should display a spinner when in the loading state', () => {
  initialState.loading = true;
  subject = shallow(<LoginForm {...initialState} />);
  expect(subject.find('Spinner').exists()).toBeTruthy();
});
