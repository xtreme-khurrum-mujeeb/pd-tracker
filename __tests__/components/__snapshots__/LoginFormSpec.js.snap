// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the LoginForm 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 0.5,
      "height": null,
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 165,
            "resizeMode": "contain",
            "width": 165,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.5,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(3,100,87,0.60)",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "paddingHorizontal": 50,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "height": 40,
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "paddingHorizontal": 7,
                }
              }
            >
              <Image
                resizeMode="contain"
                source={1}
                style={
                  Object {
                    "height": 21,
                    "width": 21,
                  }
                }
              />
            </View>
            <TextInput
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Username"
              placeholderTextColor="#F3EFEF"
              secureTextEntry={undefined}
              style={
                Object {
                  "borderBottomColor": "white",
                  "borderBottomWidth": 1,
                  "color": "white",
                  "flex": 1,
                  "fontFamily": "Avenir Next",
                  "fontSize": 20,
                  "paddingHorizontal": 10,
                }
              }
              underlineColorAndroid="transparent"
              value="test@email.ca"
            />
          </View>
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "flexDirection": "row",
                "height": 40,
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "paddingHorizontal": 7,
                }
              }
            >
              <Image
                resizeMode="contain"
                source={1}
                style={
                  Object {
                    "height": 21,
                    "width": 21,
                  }
                }
              />
            </View>
            <TextInput
              autoCorrect={false}
              onChangeText={[Function]}
              placeholder="Password"
              placeholderTextColor="#F3EFEF"
              secureTextEntry={true}
              style={
                Object {
                  "borderBottomColor": "white",
                  "borderBottomWidth": 1,
                  "color": "white",
                  "flex": 1,
                  "fontFamily": "Avenir Next",
                  "fontSize": 20,
                  "paddingHorizontal": 10,
                }
              }
              underlineColorAndroid="transparent"
              value="password123"
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 5,
                  "marginVertical": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#008774",
                    },
                    Object {
                      "fontFamily": "Avenir Next",
                      "fontSize": 20,
                    },
                  ]
                }
              >
                 
                Sign In
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 5,
                  "marginVertical": 10,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#008774",
                    },
                    Object {
                      "fontFamily": "Avenir Next",
                      "fontSize": 20,
                    },
                  ]
                }
              >
                 
                Register
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Image>
`;
